### ISSUE TITLE
Invariant Violation: Calling synchronous methods on native modules is not supported in Chrome.

### ISSUE THREAD

**Issue by irohitb** [react-native-webrtc/react-native-webrtc#1048]
I updated to version 1.92.0 from 1.89.1 (working on iOS) and I am getting following warning when I debug react-native app
```
index.bundle?platform=ios&dev=true&minify=false:48831 0 Invariant Violation: Calling synchronous methods on native modules is not supported in Chrome.

 Consider providing alternative methods to expose this method in debug mode, e.g. by exposing constants ahead-of-time.
    at invariant (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:1899:17)
    at MessageQueue.callNativeSyncHook (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:3030:11)
    at Object.nonPromiseMethodWrapper [as peerConnectionInit] (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:2724:32)
    at new RTCPeerConnection (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:109188:20)
    at getNativeRtpCapabilities$ (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:167274:22)
    at tryCatch (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:26662:19)
    at Generator.invoke [as _invoke] (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:26835:24)
    at Generator.next (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:26705:23)
    at tryCatch (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:26662:19)
    at invoke (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:26735:22)
```

#### Expected behavior

I shouldn't not be getting this warning. 

Ps: It works fine if you aren't debugging

#### What have I tried to fix it

1. Removed node_modules and installed again 
2. Remove pods folder and podlock file 
3. npm reset cache

#### Platform information

* **React Native version**: 0.62.2
* **Plugin version**: 1.92.0
* **OS**:  Tested on iOS
* **OS version**:

**Comment by saghul** [comment]
This is unfortunately a limitation we cannot fix. It's not possible to provide the proper API without using a sync method, and thus that means that using the Chrome debugger is no longer possible.

**Comment by irohitb** [comment]
@saghul how do you guys debug your application? like if you could think of any alternative and suggests accordingly?

**Comment by saghul** [comment]
I think Flipper works in this scenario.

**Comment by Romick2005** [comment]
Hi @irohitb,

[Improving data channel communication](https://github.com/react-native-webrtc/react-native-webrtc/commit/1fbe7e14bf540e1651c98ee11fc11f690f61f622#diff-489cc4b7a345533e25f69fb413978a304eed58db04a12479ebed73168d5197d4) cause this issue.

As there are not so important updates and not so significant improvements I would recommend you to downgrade to react-native-webrtc@1.89.1 for debugging. For production/staging it will work also, but if you want to can have the latest version for production/staging.

**Comment by wilau2** [comment]
This is supposed to be fixed in RN 0.66 
https://github.com/facebook/react-native/issues/32197#issuecomment-926869783

We will try to confirm here when we update.

**Comment by Romick2005** [comment]
It is not fixed.

**Comment by Jashtezakonshem** [comment]
I just created a new react-native project (v0.66.3) and still not Fixed. 
Downgrading to react-native-webrtc@1.89.1 worked for me. Thank you @Romick2005

**Comment by Romick2005** [comment]
> Please do not follow random advice which tells you to downgrade. You may expose your users to security issues, we don't support old releases.

Please provide a normal workable chrome debugging solution, then there will be no such advise. For debugging we have to use 1.89.1, but for prod I recommend to use latest. I still do not understand why this library require initial sync operations that is blocking debugging? Please fix it if you can do it or follow practical advise on random comments.

**Comment by 8BallBomBom** [comment]
@Romick2005 The frustration is understandable but you have to bare in mind no one here has to actually be here as such.
Contributions are very welcome and appreciated.

**Comment by irohitb** [comment]
@Romick2005 did you try flipper?

**Comment by Romick2005** [comment]
Yes and it works, but it seems like a huge overhead.

**Comment by 8BallBomBom** [comment]
Might feel a big chunky but in comparison to using nothing for debugging or something like android studio.
It isn't exactly that much of an overhead.

**Comment by Romick2005** [comment]
@8BallBomBom you can always have `console.log` for nothing.

**Comment by 8BallBomBom** [comment]
We're not here to argue useless points.
You do at least have an alternative debugging solution which serves the same purpose rather than nothing.
But as said previously, you are welcome to contribute.

Maybe check out the [new architecture](https://reactnative.dev/blog/2022/06/16/resources-migrating-your-react-native-library-to-the-new-architecture) coming into play, help convert the module? maybe also resolve the issue?
Simply put... being rude and demanding things when the contributors here pretty much help in their spare time isn't really the way to go about things, best to appreciate what you have and help if you can.

**Comment by eramudeep** [comment]
facing same issue on 
```
"react": "17.0.2",
 "react-native": "0.68.2",
"react-native-webrtc": "^106.0.0-beta.5"
```

**any workaround?**

**Comment by saghul** [comment]
Not really. The Chrome debugger cannot be used when synchronous methods exist, and we need those to match the WebRTC web API.

**Comment by eramudeep** [comment]
> Not really. The Chrome debugger cannot be used when synchronous methods exist, and we need those to match the WebRTC web API.

Thanks for the quick response. 
```
const startWebcam = async () => {
        pc.current = new RTCPeerConnection(servers);
        const local = await mediaDevices.getUserMedia({
            video: true,
            audio: true,
        });
        console.log("pc.current",pc.current);
        pc.current.addStream(local);

        //options.stream.getTracks().forEach(track => peerConnection.addTrack(track, options.stream))


        setLocalStream(local);
        const remote = new MediaStream();
        setRemoteStream(remote);

        // Push tracks from local stream to peer connection
        local.getTracks().forEach(track => {
            console.log(pc.current.getLocalStreams());
            pc.current.getLocalStreams()[0].addTrack(track);
        });

        // Pull tracks from remote stream, add to video stream
        pc.current.ontrack = event => {
            event.streams[0].getTracks().forEach(track => {
                remote.addTrack(track);
            });
        };

        pc.current.onaddstream = event => {
            setRemoteStream(event.stream);
        };

        setWebcamStarted(true);
    };
```

> Now i am getting addStream is not a function

**Comment by saghul** [comment]
That's because you are using the beta version. The API has changed to match Unified Plan, please check the release notes.

**Comment by eramudeep** [comment]
Thank you, let me explore the docs
